server/src/main/java/org/elasticsearch/search/rescore/RescorePhase.java

```diff
@@ -44,6 +44,15 @@ public static void execute(SearchContext context) {
         }
         try {
             for (RescoreContext ctx : context.rescore()) {
+                if (ctx.canCombineScores() == false) {
```

Comment:
ℹ️ If score can't be combined, we truncate the top docs to the window size and the rescroer will proceed to rescoring for the full top docs.

Reply:
This doesn't seem wise. If `windowSize` is smaller than `from+size` we will truncate query result.

Reply:
The truncation is now in the LearningToRankRescorer.

> This doesn't seem wise. If windowSize is smaller than from+size we will truncate query result.

We have a validation to prevent this case to happen and we check that window size is always >= from + size in the context of LTR rescorer.

We had a lot of discussion on this specific point with @jimczi and we think it is not possible to sort the TopDocs if it contains scores from the rescorer in the head and score from the first-pass in the tail.

To ensure all scores are from LTR model, we decided to truncate it to the window_size and implement the following limitations on window size:
- window_size is mandatory
- window_size is always >= from + size
- all subsequent rescorers window_size has to be smaller